app/lib/package/backend.dart

```diff
@@ -898,10 +898,7 @@ Future<_ValidatedUpload> _parseAndValidateUpload(
 
   final packageKey = db.emptyKey.append(Package, id: pubspec.name);
 
-  final versionString = canonicalizeVersion(pubspec.version);
-  if (versionString == null) {
-    throw InvalidInputException.canonicalizeVersionError(pubspec.version);
-  }
+  final versionString = pubspec.canonicalVersion;
```

Comment:
Are you sure we want to drop this? Isn't it better to reject than to have an internal error later?

Reply:
Good point, fixed.